### PR TITLE
[Bugfix][MRV2] Match fast-prefill padding to active LoRA batches

### PR DESCRIPTION
**Pull request by waizuichougou**
## Purpose

Fixes a Model Runner V2 fast-prefill regression for batches containing LoRA
requests.

The main model dispatch and the fast-prefill dispatch must use the same
`num_active_loras` value for a batch. This keeps the LoRA specialization
consistent between the two paths; their token counts may differ because
fast-prefill dispatches a reduced logits batch.

The main model dispatch computes the number of active LoRA adapters, but the
second dispatch in `FastPrefillHelper.prepare()` always used
`num_active_loras=0`. A batch with an adapter could therefore select a
no-LoRA CUDA graph descriptor for fast-prefill, even when the main model path
selected an eager or LoRA-specialized descriptor. The resulting padded logits
metadata could have a different token shape from the actual model input.

Pass the active-LoRA count from the main dispatch into fast-prefill so the
reduced logits dispatch uses the same LoRA specialization and computes padding
for its own token count. The production fix is limited to three worker files
(`+14/-3` lines); the remaining changes are regression coverage (`+61` test
lines).

## Test Plan

- Add a regression test that simulates a captured no-LoRA graph and an eager
  active-LoRA batch, verifying that fast-prefill forwards the active-LoRA
  count and does not add unrelated padding.
- Exercise 42 combinations of logits token counts (1 through 128) and active
  LoRA counts (1, 2, and 4).
- Run `tests/v1/worker/test_attn_utils.py` and
  `tests/v1/cudagraph/test_cudagraph_manager.py`.
- Run an end-to-end Gemma 4 E2B mixed batch with one base-model request and one
  rank-8 LoRA request while fast-prefill and CUDA graphs are enabled.

## Test Result

Before the fix, the regression setup selected an 8-token no-LoRA descriptor
for an active-LoRA batch whenever the no-LoRA graph was available. Across 14
token counts and 3 active-LoRA counts (42 combinations), the old choice
differed from the active-LoRA choice in 39 cases. With the fix, all 42 cases
forward the active-LoRA count and match the main dispatch's LoRA
specialization.

Tests pass:

```
70 passed, 14 warnings
```

Ruff and whitespace checks pass.

The Gemma 4 E2B mixed-batch run completed CUDA graph capture and generated
outputs for both requests (one without an adapter and one with the dummy
rank-8 adapter). This exercises model initialization, LoRA registration,
request batching, fast-prefill preparation, and graph dispatch together.

The regression is a dispatch/metadata shape error, so the worker test checks
the `BatchExecutionDescriptor` contract directly while the mixed-batch run
verifies execution through the real model path.

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://docs.vllm.ai/en/latest/contributing>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Review by njhill — APPROVED** [review]
Thank you @waizuichougou
